js_modules/dagit/packages/core/src/workers/util.ts

```diff
@@ -0,0 +1,24 @@
+// https://developer.chrome.com/blog/how-to-convert-arraybuffer-to-and-from-string/
+
+export function arrayBufferToString(buf: ArrayBuffer): string {
+  return String.fromCharCode.apply(null, new Uint16Array(buf) as any);
+}
+
+// Chunk the data into 16kb bytes to avoid hitting max argument errors when converting back to a string
+// with String.fromCharCode.apply
+const MAX_BUFFER_SIZE = 16000;
+export function stringToArrayBuffers(str: string): ArrayBuffer[] {
+  const buffers: ArrayBuffer[] = [];
+  let buf = new ArrayBuffer(0);
+  let view = new Uint16Array(buf);
+  for (let i = 0, strLen = str.length; i < strLen; i++) {
+    const index = i % MAX_BUFFER_SIZE;
+    if (index === 0) {
+      buf = new ArrayBuffer(Math.min(MAX_BUFFER_SIZE, str.length - i) * 2); // 2 bytes for each char\
+      view = new Uint16Array(buf);
+      buffers.push(buf);
+    }
+    view[index] = str.charCodeAt(i);
+  }
+  return buffers;
+}
```

Comment:
The commented URL looks fairly old and has an update note -- is there a more recent native API we can use for this?

Reply:
This is using native APIs underneath, it's just chunking to avoid going over the max number of arguments allowed for a function (Since we call `String.fromCharCode.apply` on the buffer, the number of arguments is the size of the array).